(2) Vox Populi/LUA/UnitPanel.lua

```diff
@@ -1482,18 +1483,20 @@ function TipHandler( control )
 		
 		strActionHelp = "";
 		
-		-- Add spacing for all entries after the first
-		if (bFirstEntry) then
-			bFirstEntry = false;
-		elseif (not bFirstEntry) then
+		local iGoldToScrap = unit:GetScrapGold();
+		if (iGoldToScrap > 0) then
 			strActionHelp = strActionHelp .. "[NEWLINE]";
+			strActionHelp = strActionHelp .. Locale.ConvertTextKey("TXT_KEY_SCRAP_HELP", iGoldToScrap);
 		end
 		
-		strActionHelp = strActionHelp .. "[NEWLINE]";
-		
-		local iGoldToScrap = unit:GetScrapGold();
-		
-		strActionHelp = strActionHelp .. Locale.ConvertTextKey("TXT_KEY_SCRAP_HELP", iGoldToScrap);
+		if (bDisabled) then
+			strActionHelp = strActionHelp .. "[NEWLINE][NEWLINE]";
+			if (g_UnitDeleteDisabled == 1) then
```

Comment:
Just wondering why not?
`
local textKey = "TXT_KEY_UNIT_DELETE_DISABLED_" .. (g_UnitDeleteDisabled == 1 and "GAME_OPTION" or "ENEMIES")
strActionHelp = "[COLOR_WARNING_TEXT]" .. Locale.ConvertTextKey(textKey) .. "[ENDCOLOR]"
`

Reply:
Sometimes I think about adding some utility libraries for VP. For example for text editing to abstract coloring or translations.
Kinda pseudo object.

But I do not know what is the order of lua files being loaded and what quirks it has.

```lua
local TextUtil = {}

function TextUtil.warn(text) ... end
function TextUtil.warnKey(textKey) ... end
function TextUtil.info(text) ... end
function TextUtil.infoKey(textKey) ... end
function TextUtil.error(text) ... end
function TextUtil.errorKey(textKey) ... end -- I'm not sure, is there error, warn, info, debug or coloring variants
```

Reply:
It's better to not split the text keys. One way of finding where something's implemented in the code is searching for in-game text in Localization-Merged.db and then searching the code files for the corresponding key.

Reply:
nevertheless?

```lua
local textKey =  g_UnitDeleteDisabled == 1 
    and "TXT_KEY_UNIT_DELETE_DISABLED_GAME_OPTION" 
    or "TXT_KEY_UNIT_DELETE_DISABLED_ENEMIES";
 strActionHelp = "[COLOR_WARNING_TEXT]" .. Locale.ConvertTextKey(textKey) .. "[ENDCOLOR]";
```

Reply:
yes that's better